### PR TITLE
DON-1025: Add stubbed out RegularGiving/GetAllForUser route

### PR DESCRIPTION
**Pull request by bdsl**
The main thing here is the start of working out what a regular giving mandate looks like as a JSON model to be served to front end for the donor to view. I'm sure we'll refine it as we go but worth spending time working out what we think we'll need now. Probably easiest to view and discuss it in the test file `ListRegularGivingMandatesTest.php`.

For now the data is all hard coded dummy data. Once we've decided what we want it to look like I'll go ahead and implement the mandate as an entity to load with Doctrine.

### CODE REVIEW

File: integrationTests/ListRegularGivingMandatesTest.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                    'type' => 'monthly',
+                    'dayOfMonth' => 31, // i.e. donation to be taken on last day of each calendar month
+                    'activeFrom' => '2024-08-06T00:00:00+00:00',
+                ],
+                // I'm not sure if we'll details of the donor like their name and home address
+                // recorded as part of the mandate. I think probably not - we have that on the person
+                // record and on each individual donation.
+                'charityName' => 'Some Charity',
+                'createdTime' => '2024-08-06T00:00:00+00:00',
+                'giftAid' => true, // not 100% sure yet if GA will be an option on regular giving, assuming it will.
```
Comment (bdsl) [inline integrationTests/ListRegularGivingMandatesTest.php:56]:
```suggestion
                'giftAid' => true,
```

**Comment by bdsl** [comment]
Result of running the auto generated migration. I'm not sure why it doesn't seem to include the unique constraint for combination of donor id and campaign ID that I included in the entity class.

image

File: integrationTests/ListRegularGivingMandatesTest.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                'charityName' => 'Some Charity',
+                'giftAid' => true,
+                'status' => 'active',
+                'tipAmount' => [
+                    // todo before calling ticket done: confirm if we are taking tips like this for regular giving
+                    'amountInPence' => 100,
+                    'currency' => 'GBP',
+                ],
+                'createdTime' => '2024-08-06T00:00:00+00:00',
+                'updatedTime' => '2024-08-06T00:00:00+00:00',
```
Comment (bdsl) [inline integrationTests/ListRegularGivingMandatesTest.php:58]: we also want a field for payment method as noted on Jira, but I'll probably leave that for a future PR now.

**Review by dorota-joanna — APPROVED** [review]
LGTM

**Pull request merged**